cmake: Fix compiler flags from FindWayland and FindPipeWire

Arthus opened the pull request:
${WAYLAND_DEFINITIONS} is set to  "-I/usr/include/wayland" on my system

Setting it as INTERFACE_COMPILE_DEFINITIONS leads to CMake emitting "-D-I/usr/include/wayland" in the compiler flags
this breaks the compilation

Instead add WAYLAND_DEFINITIONS as compile option so it just gets appended without adding the superflours "-D"

<!--- Please fill out the following template, which will help other contributors review your Pull Request. -->

<!--- Make sure you’ve read the contribution guidelines here: https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst -->

### Description
<!--- Describe your changes in detail. -->
<!--- If this change includes UI elements, please include screenshots. -->

### Motivation and Context
I was trying to compile OBS for the first time using openSUSE Tumbleweed. I was able to set the dependencies and cmake up, but during compilation GCC throws an error:

> \<command-line>: error: macro names must be identifiers

This is caused by cmake emitting
`cd /home/user/Programming/OBS/obs-studio/build/libobs && /usr/bin/cc -D-I/usr/include/wayland -DENABLE_DARRAY_TYPE_TEST [...]`
-D makes gcc wait for a macro, but instead the include flag follows immediately.

To fix it we just have to tell cmake to not interpret WAYLAND_DEFINITIONS as macro definition, but only append it to the compiler flags. (This is inline with what is done in [FindPipeWire.cmake.](https://github.com/obsproject/obs-studio/blob/ba68eda590b2621b13691a40bb9541ebb353257d/cmake/Modules/FindPipeWire.cmake#L89))

This issue was likely introduced by commit df94b02075636dc0ace7b4d1e9706abeb64dfbbd from PR #5155

P.S.: First day of working with OBS and first patch. I've tried following all contributor guidelines, but please tell me if I could've done better. :)

### How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment (hardware, OS version, etc.),-->
<!--- and the tests you ran, including how it may affect other areas of code. -->
With this commit I am able to compile OBS using openSUSE Tumbleweed on a Thinkpad P14s.
Snapshot 20220322 with CMake 3.22.2
Snapshot 20220323 with CMake 3.22.3
I'm able to launch the compiled binary on X11 and Wayland using KDE 5.24.3, do some simple screen capture using XSHM and Pipewire, record my desktop and click through menus.

### Types of changes
<!--- What types of changes does your PR introduce? Uncomment all that apply -->
- Bug fix (non-breaking change which fixes an issue)
<!--- - New feature (non-breaking change which adds functionality) -->
<!--- - Tweak (non-breaking change to improve existing functionality) -->
<!--- - Performance enhancement (non-breaking change which improves efficiency) -->
<!--- - Code cleanup (non-breaking change which makes code smaller or more readable) -->
<!--- - Breaking change (fix or feature that would cause existing functionality to change) -->
<!--- - Documentation (a change to documentation pages) -->

### Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

PatTheMav commented:
The fix is correct insofar as `WAYLAND_DEFINITIONS` contains the `CFLAGS` as reported by `pkg-config`.

I'd prefer if the variable is renamed as well, to avoid the confusion introduced by the fix ("definitions" being set as compiler options).

PatTheMav reviewed (changes requested):
Please also rename `WAYLAND_DEFINITIONS` to something like `WAYLAND_COMPILE_FLAGS`.

Arthus commented:
I've renamed WAYLAND_DEFINITIONS as requested and also added a commit to rename PIPEWIRE_DEFINITIONS as all your comments also apply to that one.

Question: 
Is it ok to have two commits for that, should I squash them or even open another PR?

PatTheMav commented:
Multiple commits is fine, but I'd suggest splitting the changes between one commit for Wayland and another for PipeWire.

Arthus commented:
Hi, sorry, just force pushed the commits again after some fiddling with git.

It's just as you've proposed: One commit makes the changes for pipewire. The other one fixes the compilation with wayland and also renames the wayland variable. :)

PatTheMav commented:
Minor nit to pick: One commit explicitly mentions Wayland, the other doesn't. I'd suggest renaming the commit to explicitly call out that it changes something related to PipeWire.

Arthus commented:
Done :)

RytoEX reviewed (changes requested):
Minor nitpicks on the commit messages.

Use a capitalized word immediately after the commit message prefix. Punctuate paragraphs in the commit message body correctly.

The first commit doesn't seem to have anything to do directly with libobs, so mentioning it is confusing. The changes are just in our CMake finders, so I don't think it makes sense to mention libobs, unless you can elaborate?

RytoEX reviewed (approved):
I've adjusted the commit messages and the PR title. Merging shortly.
